pm/sendermonitor.go

```diff
@@ -19,28 +22,24 @@ var unixNow = func() int64 {
 // SenderMonitor is an interface that describes methods used to
 // monitor remote senders
 type SenderMonitor interface {
-	RedeemableEmitter
-
 	// Start initiates the helper goroutines for the monitor
 	Start()
 
 	// Stop signals the monitor to exit gracefully
 	Stop()
 
 	// QueueTicket adds a ticket to the queue for a remote sender
-	QueueTicket(addr ethcommon.Address, ticket *SignedTicket) error
-
-	// AddFloat adds to a remote sender's max float
-	AddFloat(addr ethcommon.Address, amount *big.Int) error
-
-	// SubFloat subtracts from a remote sender's max float
-	SubFloat(addr ethcommon.Address, amount *big.Int)
+	QueueTicket(ticket *SignedTicket) error
 
 	// MaxFloat returns a remote sender's max float
 	MaxFloat(addr ethcommon.Address) (*big.Int, error)
 
 	// ValidateSender checks whether a sender's unlock period ends the round after the next round
 	ValidateSender(addr ethcommon.Address) error
+
+	// MonitorMaxFloat listens for max floats updates for a sender
+	// This method only needs to be "implemented" when the SenderMonitor consumer is a fire-and-forget module for winning tickets
+	MonitorMaxFloat(sender ethcommon.Address, sink chan<- *big.Int) event.Subscription
```

Comment:
Does the sink need to be supplied by the caller, or can it be created and returned from the function? Like this:

```golang
MonitorMaxFloat(sender ethcommon.Address) (sink chan<- *big.Int, event.Subscription)
```

Reply:
Having the caller supply the channel makes unit testing a little easier and we have the caller supply the channel for all subscriptions in the repo so it made sense to be consistent.